### PR TITLE
[server] Remove Workspace Factory EE, merge with standard

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Merges EE and non-EE WorkspaceFactory, we no longer support EE

## Related Issue(s)

Fixes #

## How to test

1. Preview

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-c7c0ae30cf</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-c7c0ae30cf.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-c7c0ae30cf.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/workspace/workspace-factory.ts [hunk 1]
```diff
@@ -44,6 +53,12 @@ export class WorkspaceFactory {
         context: WorkspaceContext,
         normalizedContextURL: string,
     ): Promise<Workspace> {
+        if (StartPrebuildContext.is(context)) {
+            return this.createForStartPrebuild(ctx, user, organizationId, context, normalizedContextURL);
+        } else if (PrebuiltWorkspaceContext.is(context)) {
+            return this.createForPrebuiltWorkspace(ctx, user, organizationId, project, context, normalizedContextURL);
+        }
```
Comment (easyCZ) [inline components/server/src/workspace/workspace-factory.ts:60]: The placement here is exactly matching the superclass invocation chain. However, I believe it should come after the   following check
```
if (user.additionalData?.isMigratedToTeamOnlyAttribution && !organizationId) {
            throw new ResponseError(ErrorCodes.INVALID_VALUE, "Cannot create workspace without organization.");
        }
```
@svenefftinge could you confirm, please?

**Comment by easyCZ** [comment]
> Code LGTM. [This comment](https://github.com/gitpod-io/gitpod/pull/17339#discussion_r1175091141) looks like an improvement. ✔️

Applied the change, looks like this was just an over-sight due to our split brain between EE and non.

**Comment by geropl** [comment]
> Applied the change, looks like this was just an over-sight due to our split brain between EE and non.

No, it isn't. It's just not there, yet. So far, the whole "org-only" change is mostly frontend only. :+1:

File: components/server/src/workspace/workspace-factory.ts (hunk 1 above)
Comment (svenefftinge) [reply components/server/src/workspace/workspace-factory.ts:64]: Yes, that looks correct

**Pull request merged**